src/vulkan-renderer/vk_tools/representation.cpp

```diff
@@ -742,13 +752,13 @@ std::string memory_property_flag_to_string(const VkMemoryPropertyFlags bit) {
     default:
         break;
     }
-
-    // Don't forget to offer a default value.
     return "Unknown";
 }
 
-std::string memory_heap_flag_to_string(const VkMemoryHeapFlagBits bit) {
-    switch (bit) {
+/// @brief Convert a VkMemoryHeapFlagBits value into the corresponding value as std::string.
+/// @param bit The memory heap flag bit.
+std::string_view memory_heap_flag_to_string(const VkMemoryHeapFlagBits heap_flag_bit) {
```

Comment:
I think you forgot to specialize this one?

Reply:
It's interesting that this does compile without this template specialization as well:
```cpp
spdlog::debug("{}", vk_tools::as_string<VkMemoryHeapFlags>(mem_heap_prop_flag));
```

as it can probably do a conversion from `VkMemoryPropertyFlags` to `VkMemoryHeapFlags`.

Reply:
Done.